adapters/celestia/src/celestia.rs

```diff
@@ -257,20 +263,18 @@ impl BlockHeader for CelestiaHeader {
     type Hash = TmHash;
 
     fn prev_hash(&self) -> Self::Hash {
-        // Try to return the cached value
-        if let Some(hash) = self.cached_prev_hash.borrow().as_ref() {
+        let cached_hash = self.cached_prev_hash.get();
+        if let Some(hash) = cached_hash {
             return hash.clone();
         }
-        // If we reach this point, we know that the cach is empty - so there can't be any outstanding references to its value.
-        // That means its safe to borrow the cache mutably and populate it.
-        let mut cached_hash = self.cached_prev_hash.borrow_mut();
+
         let hash =
             <tendermint::block::Id as Protobuf<celestia_tm_version::types::BlockId>>::decode(
                 self.header.last_block_id.as_ref(),
             )
             .expect("must not call prev_hash on block with no predecessor")
             .hash;
-        *cached_hash = Some(TmHash(hash));
+        self.cached_prev_hash.set(TmHash(hash));
```

Comment:
It looks like `get_or_init()` is made for exactly this. Should we use it instead?

Reply:
Good catch!